### PR TITLE
2021.81

### PR DESCRIPTION
**Pull request by janettelynch**
Release notes draft.

### CODE REVIEW

**Review by aiwilliams — APPROVED** [review]

**Review by erkangz — APPROVED** [review]

**Review by AnneNielsenJ1 — COMMENTED** [review]
@ninetreats Can you please review the updates I made with the GitHub secrets thing?

File: release-notes/2021-81.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+J1 will be deploying an updated login experience in the next few weeks - stay tuned!
+
+![](../assets/new_login_screen.png)
+
+## New Features and Improvements
+
+- Delivered a refreshed landing page for querying your data.
+
+- Delivered a new onboarding To-Do list for new J1 users to get oriented and obtain
```
Comment (austinkelleher) [inline release-notes/2021-81.md:16]: Would be awesome to see a screenshot of this in the release notes!

File: release-notes/2021-81.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+### GitHub
+
+- With this update, JupiterOne has requested read-only access to Github Organization Secrets, 
+  Github Repository Secrets, and environment secrets metadata which must be granted in the JupiterOne app in Github. 
+  Read-only access allows us to access information about the secrets, but not the encrypted values. 
+  We have no access to the values of secrets at any time, nor will JupiterOne request such access.
+  
+  This access will allow you to ask questions in JupiterOne about your secrets, such as 
+  "What Secrets are older than 365 days?" or "Which Repos have access to Org Secrets?" or
```
Comment (austinkelleher) [inline release-notes/2021-81.md:54]: Interesting! It seems that we do not have managed questions for these. I'll make sure these gets added as managed questions. I wonder if we should put actual J1QL queries/questions in the release notes for users to try!

File: release-notes/2021-81.md (hunk 1 above)
Comment (austinkelleher) [reply release-notes/2021-81.md:16]: Also, should "To-Do" be capitalized?

File: release-notes/2021-81.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+  - Get an organization secret: https://docs.github.com/en/rest/reference/actions#get-an-organization-secret
+  
+  - Get a repository secret: https://docs.github.com/en/rest/reference/actions#get-a-repository-secret
+
+  - Get an environment secret: https://docs.github.com/en/rest/reference/actions#get-an-environment-secret 
+
+  If an existing GitHub integration user does not grant these additional permissions in their GitHub account, 
+  then JupiterOne will automatically skip these steps in the integration and not import the Organization and 
+  Repository Secret Entities and Properties. The rest of data for already approved permissions will continue
```
Comment (austinkelleher) [inline release-notes/2021-81.md:68]: Wondering whether these words should be lowercase in this paragraph:

- "Repository"
- "Secret"
- "Entities"
- "Properties"
- "Organization"

File: release-notes/2021-81.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+- Improved OAuth authorization error handling. The integration now reports
+  proper error messaging when Heroku responds to HTTP status codes that indicate
+  an error occurred.
+
+### Tenable.io
+
+- Added the `name` property to `tenable_vulnerability_finding` entities.
+
+- Fixed the disabled `buildAssetVulnerabilityRelationships` step.
```
Comment (austinkelleher) [inline release-notes/2021-81.md:160]: Hey @ndowmon can you elaborate on what this means for customers?

File: release-notes/2021-81.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+- Fixed duplicate keys for `tenable_vulnerability_finding_is_cve` mapped 
+  relationships
+
+- Fixed duplicate keys for `tenable_asset` entities.
+
+- Fixed duplicate keys for `tenable_vulnerability_finding` entities.
+
+- Fixed `Cannot read property 'map' of undefined` when iterating CVEs for `tenable_vulnerability_finding`.
```
Comment (austinkelleher) [inline release-notes/2021-81.md:169]: I wonder if we should rephrase this to something like:

> "Fixed issue that prevented complete set of `tenable_vulnerability_finding` from being ingested"

Comment (austinkelleher) [reply release-notes/2021-81.md:169]: I think the same comment can be applied for other instances of "duplicate keys" above.

File: release-notes/2021-81.md (hunk 4 above)
Comment (ndowmon) [reply release-notes/2021-81.md:160]: See https://github.com/JupiterOne/graph-tenable-cloud/pull/136. This was actually not ever enabled.

Let's change to:

```suggestion
- Enable the `buildAssetVulnerabilityRelationships` step.
```

Comment (austinkelleher) [reply release-notes/2021-81.md:160]: @ndowmon Thanks, but can you elaborate on what the actual outcome is for the customer?

The following is pretty technically specific to our implementation:

> Enable the `buildAssetVulnerabilityRelationships` step.

Perhaps something like:

```md
- Added support for ingesting the following new relationships:
  
| Source Entity `_type` | Relationship `_class` | Target Entity `_type`                       |
| --------------------- | --------------------- | -------------------------------        |
| `tenable_asset`       | **HAS**               | `tenable_vulnerability_finding`                  |
```

File: release-notes/2021-81.md (hunk 2 above)
Comment (janettelynch) [reply release-notes/2021-81.md:56]: @AnneNielsenJ1 What do you think?

File: release-notes/2021-81.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+## Announcements
+
+J1 will be deploying an updated login experience in the next few weeks - stay tuned!
+
+![](../assets/new_login_screen.png)
+
+## New Features and Improvements
+
+- A refreshed landing page for querying your data. The updated search app also
```
[the remark this answers is not in the document]
Comment (janettelynch) [reply release-notes/2021-81.md]: Will have to hold off as it is not finalized.

File: release-notes/2021-81.md (hunk 2 above)
Comment (erkangz) [reply release-notes/2021-81.md:56]: Since these questions are mentioned here, it makes sense to add at least these two as managed questions to the library. Takes 5 mins to do.

File: release-notes/2021-81.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+J1 will be deploying an updated login experience in the next few weeks - stay tuned!
+
+![](../assets/new_login_screen.png)
+
+## New Features and Improvements
+
+- A refreshed landing page for querying your data. The updated search app also 
+  includes an onboarding To-Do list for new J1 users to help them get oriented and obtain 
+  value from their data.
+- You can now perform bulk uploads of data in the UI without a file size limit.
```
[the remark this answers is not in the document]
Comment (AnneNielsenJ1) [reply release-notes/2021-81.md]: Good feedback - thank you! I also add a DOC ticket to document how bulk upload works.
